chunk/storage_client.go

```diff
@@ -31,3 +34,33 @@ type ReadBatch interface {
 	RangeValue(index int) []byte
 	Value(index int) []byte
 }
+
+// StorageClientConfig chooses which storage client to use.
+type StorageClientConfig struct {
+	StorageClient string
+	AWSStorageConfig
+}
+
+// RegisterFlags adds the flags required to configure this flag set.
+func (cfg *StorageClientConfig) RegisterFlags(f *flag.FlagSet) {
+	flag.StringVar(&cfg.StorageClient, "chunk.storage-client", "aws", "Which storage client to use (aws, inmemory).")
+	cfg.AWSStorageConfig.RegisterFlags(f)
+}
+
+// NewStorageClient makes a storage client based on the configuration.
+func NewStorageClient(cfg StorageClientConfig) (StorageClient, string, error) {
+	switch cfg.StorageClient {
+	case "inmemory":
+		return NewMockStorage(), "", nil
+	case "aws":
+		// TODO(jml): Remove this once deprecation period expires - 2017-03-21.
```

Comment:
Deprecation period is just until we have this in prod, right?  @Bplotka this could break your deploys when we remove this flag - are you guys depending on Cortex yet?